AdminRouter/start-windows-build.ps1

```diff
@@ -0,0 +1,266 @@
+            $parameters['FilePath'] = 'msiexec.exe'
+            $parameters['ArgumentList'] += @("/i", $programFile)
+        }
+        Write-Output "Installing $programFile"
+        $p = Start-Process @parameters
+        if($p.ExitCode -ne 0) {
+            Throw "Failed to install prerequisite $programFile during the environment setup"
+        }
+    }
+    # Add all the tools to PATH
+    $toolsDirs = @("$GIT_DIR\cmd", "$GIT_DIR\bin", "$7ZIP_DIR")
+    $env:PATH += ';' + ($toolsDirs -join ';')
+}
+
+function Get-LatestCommitID {
+    return $global:LATEST_COMMIT_ID
+}
+
+function Set-LatestAdminRouterCommit {
+    # The following id is from https://www.apachelounge.com/download/VC15/binaries/httpd-2.4.33-win64-VC15.zip.txt
+    $commitId = "65525690B713E266A7EA44D60C0D2BCEE4716B2D"
+    Set-Variable -Name "LATEST_COMMIT_ID" -Value $commitId -Scope Global  -Option ReadOnly
+}
+
+function Get-MesosJenkinsRepo {
+    Write-Output "Get-MesosJenkinsRepo"
+    New-Directory $ADMINROUTER_DIR | Out-Null
+    New-Directory $ADMINROUTER_BUILD_OUT_DIR -RemoveExisting | Out-Null
+    New-Directory $ADMINROUTER_BUILD_LOGS_DIR  -RemoveExisting | Out-Null
+    $global:PARAMETERS["BRANCH"] = $Branch
+    Remove-Item $ADMINROUTER_MESOS_JENKINS_GIT_REPO_DIR -Force  -Recurse -ErrorAction SilentlyContinue
+
+    Write-Output "Cloning Mesos-Jenkins repo"    
+    Start-GitClone -Path $ADMINROUTER_MESOS_JENKINS_GIT_REPO_DIR -URL $MESOS_JENKINS_GIT_URL -Branch $Branch
+    Write-Output "Cloning  done"
+}
+
+function Get-ApacheServerPackage {
+    Write-Output "Downloading Apache HTTP Server zip file: $WINDOWS_APACHEL_HTTP_SERVER_URL"
+    $filesPath = Join-Path $env:TEMP "httpd-2.4.33-win64-VC15.zip"
+    Start-ExecuteWithRetry { Invoke-WebRequest -UseBasicParsing -Uri $WINDOWS_APACHEL_HTTP_SERVER_URL -OutFile $filesPath}
+
+    # To be sure that a download is intact and has not been tampered with from the original download side
+    # we need to perform biniary hash validation
+    Write-Output "Validating Hash value for downloaded file: $filesPath"
+    $hashFromDowloadedFile = Get-FileHash $filesPath -Algorithm SHA256
+    Write-Output "Expected SHA256 hash is [$WINDOWS_APACHEL_HTTP_SERVER_SHA256]"
+    Write-Output "Computed SHA256 hash is [$($hashFromDowloadedFileHash)]"
+
+    # validate both hashes are the same
+    if ($hashFromDowloadedFile.Hash -eq $WINDOWS_APACHEL_HTTP_SERVER_SHA256) {
+        Write-Host 'Hash validation test passed' -ForegroundColor Green
+    } else {
+        Write-Host 'Hash validation test FAILED!!' -ForegroundColor Red
+    }
+
+    Write-Output "Extracting binaries archive in: $ADMINROUTER_DIR"
+    Expand-Archive -LiteralPath $filesPath -DestinationPath $ADMINROUTER_DIR
+    Remove-item $filesPath
+    Write-Output "Finshed downloading"  
+    Set-LatestAdminRouterCommit
+}
+
+function New-DCOSAdminRouterPackage {
+    Write-Output "Generating DC/OS AdminRouter package $ADMINROUTER_BUILD_BINARIES_DIR\adminrouter.zip"
```

Comment:
As a suggestion it may be handy to make the string `$ADMINROUTER_BUILD_BINARIES_DIR\adminrouter.zip` a constant and add a comment on top of it

Reply:
Advice taken.